app/models/employee.rb

```diff
@@ -277,7 +272,7 @@ def self.adv_search(p)
       end
     end
 
-    if status = p[:status]
+    if status.p[:status]
```

Comment:
`.` here might change the complete logic of the condition. Please test properly.